lightning/src/ln/channelmanager.rs

```diff
@@ -5039,12 +5131,22 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> Writeable f
 		}
 
 		let pending_outbound_payments = self.pending_outbound_payments.lock().unwrap();
-		(pending_outbound_payments.len() as u64).write(writer)?;
-		for session_priv in pending_outbound_payments.iter() {
-			session_priv.write(writer)?;
+		// For backwards compat, write the session privs and their total length.
+		let mut num_pending_outbounds_compat: u64 = 0;
+		for (_, outbounds) in pending_outbound_payments.iter() {
+			num_pending_outbounds_compat += outbounds.len() as u64;
+		}
+		num_pending_outbounds_compat.write(writer)?;
+		for (_, outbounds) in pending_outbound_payments.iter() {
+			for outbound in outbounds.iter() {
+				outbound.write(writer)?;
+			}
 		}
 
-		write_tlv_fields!(writer, {});
+		let pending_outbound_payments = Some(pending_outbound_payments);
+		write_tlv_fields!(writer, {
+			(0, pending_outbound_payments, option),
```

Comment:
This doesn't need the `Some` wrapper and should probably be odd.

```suggestion
		write_tlv_fields!(writer, {
			(1, pending_outbound_payments, required),
```